include/bluetooth/mesh/light_ctrl_reg_spec.h

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+/** @file
+ *  @defgroup bt_mesh_light_ctrl_reg_spec Light Lightness Control Spec Regulator
+ *  @ingroup bt_mesh_light_ctrl
+ *  @{
+ *  @brief Light Lightness Control spec regulator
+ */
+
+#ifndef BT_MESH_LIGHT_CTRL_REG_SPEC_H__
+#define BT_MESH_LIGHT_CTRL_REG_SPEC_H__
+
+#include <bluetooth/mesh/light_ctrl_reg.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**  @def BT_MESH_LIGHT_CTRL_REG_SPEC_INIT
+ *
+ *   @brief Initialization macro for @ref bt_mesh_light_ctrl_reg_spec.
+ */
+#define BT_MESH_LIGHT_CTRL_REG_SPEC_INIT                                       \
+	{                                                                      \
+		.reg = {                                                       \
+			.init = bt_mesh_light_ctrl_reg_spec_init,              \
+			.start = bt_mesh_light_ctrl_reg_spec_start,            \
+			.stop = bt_mesh_light_ctrl_reg_spec_stop               \
+		}                                                              \
+	}
+
+/** Light Lightness Control spec regulator context. */
+struct bt_mesh_light_ctrl_reg_spec {
+	/** Common regulator context */
+	struct bt_mesh_light_ctrl_reg reg;
+	/** Regulator step timer */
+	struct k_work_delayable timer;
```

Comment:
I'm inclined to suggest that this should be a `struct k_timer` that can run in an interrupt, so we can do the entire regulator part in interrupts if we'd like. The light_ctrl_srv would have to move the updates to a `k_work` itself, so we don't push lightness callbacks in interrupts. This should also allow removal of the `enabled` flag, as `k_timer_stop` can't fail.

Reply:
Can we postpone this to whenever we extract this out of the mesh?

Reply:
Sure!